registration/scripts/directives.js

```diff
@@ -56,7 +63,13 @@
                             if (angular.isDefined(deferred)) {
                                 deferred.notify('reject', response);
                             }
+                            apiError = true;
                         });
+                    } else {
+                        //check there are no other error messages on the screen from the api
+                        if (!apiError) {
+                            scope.emptyEmail = true;
+                        }
```

Comment:
I agree there should probably be an error from the api. That said, if we were to check for an empty email in the front-end then we should do it before sending the request to the api as there is no point in sending a request we know is not valid.